### PR TITLE
treat unknown actors as internal for the purpose of audit logs

### PR DESCRIPTION
**Pull request by daxmc99**
Slack context https://sourcegraph.slack.com/archives/C1JH2BEHZ/p1688666832067379

Loom for more context: https://www.loom.com/share/85b5c2c903164344a4f27265afba16fd?sid=df673252-7b70-4e32-9fd9-01b3b74828f5

## Test plan

Small change to audit logs to reduce uneccessary logging of internal actors
We can still use the configuration `log.auditLog.internalTraffic` to enable logging of internal actors

### CODE REVIEW

**Review by mohammadualam — APPROVED** [review]

File: internal/audit/audit.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func Log(ctx context.Context, logger log.Logger, record Record) {
 
 	// internal actors add a lot of noise to the audit log
 	siteConfig := conf.SiteConfig()
-	if act.Internal && !IsEnabled(siteConfig, InternalTraffic) {
+	if (act.Internal || actorId(act) == "unknown") && !IsEnabled(siteConfig, InternalTraffic) {
```
Comment (michaellzc) [inline internal/audit/audit.go:26]: docstring to explain the purpose of the filter here

Comment (daxmc99) [reply internal/audit/audit.go:27]: Updated!

File: internal/audit/audit.go [hunk 2]
```diff
@@ -23,7 +23,8 @@ func Log(ctx context.Context, logger log.Logger, record Record) {
 
 	// internal actors add a lot of noise to the audit log
 	siteConfig := conf.SiteConfig()
-	if act.Internal && !IsEnabled(siteConfig, InternalTraffic) {
+	// if the actor is internal or unknown, and internal traffic logging is disabled, do not log
```
Comment (michaellzc) [inline internal/audit/audit.go:26]:
```suggestion
	// if the actor is internal or unknown, and internal traffic logging is disabled, do not log
	// internal actors generate a large volume of logs, and they are generally not useful
```

**Review by michaellzc — APPROVED** [review]

**Review by bobheadxi — COMMENTED** [review]
I think this might hurt efforts to ensure that all requests happen with an explicit actor, though I guess that hasn't been a major effort in a while

**Review by camdencheek — APPROVED** [review]

**Review by bobheadxi — APPROVED** [review]
Good to unblock, but might need to find an owner for actor propagation (maybe @sourcegraph/iam ?) to have a look as well

**Comment by camdencheek** [comment]
Looks like we'll need some test updates

**Comment by daxmc99** [comment]
@camdencheek Updated the test

**Comment by davejrt** [comment]
@daxmc99 your test failures got swallowed by the large log

```  sg git:(dax/disable_audit_internal) bazel test //cmd/frontend/internal/httpapi:httpapi_test
Starting local Bazel server and connecting to it...
INFO: Analyzed target //cmd/frontend/internal/httpapi:httpapi_test (1606 packages loaded, 22959 targets configured).
INFO: Found 1 test target...
INFO: Deleting stale sandbox base /private/var/tmp/_bazel_davetry/0f2ae25a7c5caa7f8eb4187de5806e28/sandbox
FAIL: //cmd/frontend/internal/httpapi:httpapi_test (see /private/var/tmp/_bazel_davetry/0f2ae25a7c5caa7f8eb4187de5806e28/execroot/__main__/bazel-out/darwin_arm64-fastbuild/testlogs/cmd/frontend/internal/httpapi/httpapi_test/test.log)
INFO: From Testing //cmd/frontend/internal/httpapi:httpapi_test:
==================== Test output for //cmd/frontend/internal/httpapi:httpapi_test:
--- FAIL: Test_recordAuditLog (0.00s)
    --- FAIL: Test_recordAuditLog/GraphQL_requests_are_audit_logged_when_audit_log_is_enabled (0.00s)
        graphql_test.go:67: 
                Error Trace:    cmd/frontend/internal/httpapi/graphql_test.go:67
                Error:          Not equal: 
                                expected: 0
                                actual  : 1
                Test:           Test_recordAuditLog/GraphQL_requests_are_audit_logged_when_audit_log_is_enabled
panic: runtime error: index out of range [0] with length 0 [recovered]
        panic: runtime error: index out of range [0] with length 0

goroutine 259 [running]:
testing.tRunner.func1.2({0x10bdb6660, 0xc000e0eea0})
        GOROOT/src/testing/testing.go:1396 +0x278
testing.tRunner.func1()
        GOROOT/src/testing/testing.go:1399 +0x41c
panic({0x10bdb6660, 0xc000e0eea0})
        GOROOT/src/runtime/panic.go:890 +0x258
github.com/sourcegraph/sourcegraph/cmd/frontend/internal/httpapi.Test_recordAuditLog.func1(0x0?)
        cmd/frontend/internal/httpapi/graphql_test.go:68 +0x980
testing.tRunner(0xc0023f0d00, 0xc0023894f0)
        GOROOT/src/testing/testing.go:1446 +0x18c
created by testing.(*T).Run
        GOROOT/src/testing/testing.go:1493 +0x560
================================================================================
Target //cmd/frontend/internal/httpapi:httpapi_test up-to-date:
  bazel-bin/cmd/frontend/internal/httpapi/httpapi_test_/httpapi_test
INFO: Elapsed time: 90.265s, Critical Path: 66.06s
INFO: 1255 processes: 5 internal, 1250 darwin-sandbox.
//cmd/frontend/internal/httpapi:httpapi_test                             FAILED in 5.4s
  /private/var/tmp/_bazel_davetry/0f2ae25a7c5caa7f8eb4187de5806e28/execroot/__main__/bazel-out/darwin_arm64-fastbuild/testlogs/cmd/frontend/internal/httpapi/httpapi_test/test.log

Executed 1 out of 1 test: 1 fails locally.
INFO: Build completed, 1 test FAILED, 1255 total actions
➜  sg git:(dax/disable_audit_internal) bazel test //cmd/gitserver/server/accesslog:accesslog_test
INFO: Analyzed target //cmd/gitserver/server/accesslog:accesslog_test (0 packages loaded, 3 targets configured).
INFO: Found 1 test target...
FAIL: //cmd/gitserver/server/accesslog:accesslog_test (see /private/var/tmp/_bazel_davetry/0f2ae25a7c5caa7f8eb4187de5806e28/execroot/__main__/bazel-out/darwin_arm64-fastbuild/testlogs/cmd/gitserver/server/accesslog/accesslog_test/test.log)
INFO: From Testing //cmd/gitserver/server/accesslog:accesslog_test:
==================== Test output for //cmd/gitserver/server/accesslog:accesslog_test:
--- FAIL: TestHTTPMiddleware (0.00s)
    --- FAIL: TestHTTPMiddleware/disabled,_then_enabled (0.00s)
        accesslog_test.go:142: 
                Error Trace:    cmd/gitserver/server/accesslog/accesslog_test.go:142
                Error:          "[{2023-07-12 20:48:11.339235 +0000 UTC m=+0.022214751 TestHTTPMiddleware/disabled,_then_enabled info access logging enabled map[]}]" should have 2 item(s), but has 1
                Test:           TestHTTPMiddleware/disabled,_then_enabled
FAIL
================================================================================
Target //cmd/gitserver/server/accesslog:accesslog_test up-to-date:
  bazel-bin/cmd/gitserver/server/accesslog/accesslog_test_/accesslog_test
INFO: Elapsed time: 2.200s, Critical Path: 1.80s
INFO: 9 processes: 3 internal, 6 darwin-sandbox.
//cmd/gitserver/server/accesslog:accesslog_test                          FAILED in 0.7s
  /private/var/tmp/_bazel_davetry/0f2ae25a7c5caa7f8eb4187de5806e28/execroot/__main__/bazel-out/darwin_arm64-fastbuild/testlogs/cmd/gitserver/server/accesslog/accesslog_test/test.log

Executed 1 out of 1 test: 1 fails locally.
INFO: Build completed, 1 test FAILED, 9 total actions```

**Comment by camdencheek** [comment]
I pushed a commit to fix the remaining tests so we can get the patch rolling. IMO, we should revisit this after the release though. It seems a little odd that we can avoid access logging by not having an actor set.

**Pull request merged**